File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer =0
    arr = [0]
    for num in numbers:
        tmp = []
        for i in arr:
            tmp.append(i + num)
            tmp.append(i - num)
        arr = tmp
    for _ in arr:
        if _ == target:
            answer += 1
    return answer

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
class Solution {
    public int solution(int[] numbers, int target) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(0);
        for (int idx=0;idx< numbers.length;idx++) {
            for (int i = 0; i < Math.pow(2,idx); i++) {
                Integer tmp = queue.poll();
                queue.add(tmp + numbers[idx]);
                queue.add(tmp - numbers[idx]);
            }

        }
        return Collections.frequency(queue,target);
    }
}